### PR TITLE
Updated referee model to broadcast last player to touch the ball

### PR DESCRIPTION
**Pull request by sanmit**
Server updates to pass player index in capture message

### CODE REVIEW

File: src/referee.cpp [hunk 1]
```diff
@@ -3099,6 +3110,14 @@ HFORef::analyse()
             {
                 M_take_time = 0;
             }
+
+            // Broadcast who has possession. Goalie catches are broadcast separately by the server. Rebroadcasting a message here would overwrite the other one. 
+            if (!(M_holder_side == 'R' && M_holder_unum == 1)){
```
Comment (mhauskn) [inline src/referee.cpp]: Let's break this into two lines: 
bool goalie_has_ball = M_holder_side == 'R' && M_holder_unum == 1;
if (!goalie_has_ball) { ...

File: src/referee.cpp [hunk 2]
```diff
@@ -3099,6 +3110,14 @@ HFORef::analyse()
             {
                 M_take_time = 0;
             }
+
+            // Broadcast who has possession. Goalie catches are broadcast separately by the server. Rebroadcasting a message here would overwrite the other one.
```
Comment (mhauskn) [inline src/referee.cpp:3114]: Do we need to broadcast every step or can we just broadcast when a new player gets possession of the ball?

File: src/referee.cpp [hunk 3]
```diff
@@ -2996,6 +2997,8 @@ HFORef::HFORef( Stadium & stadium )
       M_defense( 0 ),
       M_time( 0 ),
       M_take_time( 0 ),
+      M_holder_unum(-1),
+      M_holder_side('U'),
```
Comment (mhauskn) [inline src/referee.cpp:3001]: Add a comment explaining what 'U/L/R' stand for.

**Comment by mhauskn** [comment]
Thanks for the changes! My largest questions are:
1) Can we only broadcast a message when a new/different player gains posession of the ball?
2) Can we broadcast more than one message per cycle? If so, would it make sense to separate the end_of_trial messages from the 'gained_possession' messages?

File: src/referee.cpp (hunk 3 above)
Comment (sanmit) [inline src/referee.cpp:3001]: Added a comment to referee.h where this variable is declared

File: src/referee.cpp (hunk 2 above)
Comment (sanmit) [inline src/referee.cpp:3114]: It's 2 extra ints per cycle, which are negligible. The alternative would be to store the playerOnBall internally in referee, and send it when it changes. But then the parser in agent.cpp needs to handle both cases. I don't think it's worth the complexity.

**Pull request merged**